src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt

```diff
@@ -332,10 +343,15 @@ class RsErrorAnnotator : RsAnnotatorBase(), HighlightRangeExtension {
     }
 
     private fun checkCondition(holder: AnnotationHolder, element: RsCondition) {
-        val pat = element.pat ?: return
-        if (pat.isIrrefutable) {
+        val pat = element.pat
+        if (pat != null && pat.isIrrefutable) {
             checkFeature(holder, pat, IRREFUTABLE_LET_PATTERNS, "irrefutable let pattern")
```

Comment:
Does this feature should be checked in case `if let <refutable pattern> | <irrefutable pattern>` and `if let <irrefutable pattern> | <refutable pattern>`? If such cases are possible, let's add corresponding tests to `RsErrorAnnotatorTest.kt`

Reply:
It's an open question - I did investigate it (see comment in PR description)
I've just keep previous behavior for all possible cases before stabilization of the feature. So it can produce false positive annotations with new syntax